src/network.ts

```diff
@@ -1,27 +1,64 @@
-import { Handle } from "./core/contracts/registry";
-import { ConfigOpts } from "./config";
-import { NotImplementedError } from "./core/utilities/errors";
+import { resolveRegistration, Handle } from "./core/contracts/registry";
+import * as messages from "./core/messages/messages";
+import * as batchMessages from "./core/batch/batchMesssages";
+import * as config from "./config";
+import { UserNotFound } from "./handles";
+import { MissingUser, NotImplementedError } from "./core/utilities";
 
 /**
- * follow() creates a follow event and enqueues it for the next batch. This
- * method is not yet implemented.
+ * follow() creates a follow event and returns it.
  *
- * @param handle  The handle of the user to follow
- * @param opts    Optional. Configuration overrides, such as from address, if any
+ * @param handle The handle of the user to follow
+ * @param opts   Optional. Configuration overrides, such as from address, if any
+ * @returns      The signed DSNP Graph Change message
  */
-export const follow = async (_handle: Handle, _opts?: ConfigOpts): Promise<void> => {
-  throw NotImplementedError;
+export const follow = async (
+  handle: Handle,
+  opts?: config.ConfigOpts
+): Promise<batchMessages.BatchGraphChangeMessage> => {
+  // Get current user id
+  const { currentUserId } = await config.getConfig(opts);
+  if (!currentUserId) throw MissingUser;
+
+  // Get followeeId from handle
+  const registration = await resolveRegistration(handle);
+  if (!registration) throw UserNotFound;
```

Comment:
I thought we were moving away from User to Registration.  If we are missing a registration, surely the error should be called `RegistrationNotFound`.  I think we should pick one and stick with it.

Reply:
Renamed to `RegistrationNotFound`